tests/app/services/test_query_builder.py

```diff
@@ -60,28 +79,26 @@ def test_aggregation_throws_error_if_not_implemented():
         construct_query(build_query_params(), aggregations=['missing'])
 
 
-def test_should_make_multi_match_query_if_keywords_supplied():
+def test_should_make_multi_match_query_if_keywords_supplied(services_mapping):
     keywords = "these are my keywords"
     query = construct_query(build_query_params(keywords))
     assert_equal("query" in query, True)
     assert_in("simple_query_string", query["query"])
     query_string_clause = query["query"]["simple_query_string"]
     assert_equal(query_string_clause["query"], keywords)
     assert_equal(query_string_clause["default_operator"], "and")
-    assert_equal(set(query_string_clause["fields"]), set(TEXT_FIELDS))
+    assert_equal(set(query_string_clause["fields"]), services_mapping.text_fields_set)
 
 
-def test_should_identify_filter_search_from_query_params():
-    cases = (
-        (build_query_params(), False),
-        (build_query_params(keywords="lot"), False),
-        (build_query_params(filters={'lot': "lot"}), True),
-        (build_query_params(keywords="something", filters={'lot': "lot"}), True),
-        (build_query_params(filters={'serviceTypes': ["serviceTypes"]}), True)
-    )
-
-    for query, expected in cases:
-        yield assert_equal, is_filtered(query), expected, query
+@pytest.mark.parametrize("query,expected", (
+    (build_query_params(), False),
+    (build_query_params(keywords="lot"), False),
+    (build_query_params(filters={'lot': "lot"}), True),
+    (build_query_params(keywords="something", filters={'lot': "lot"}), True),
+    (build_query_params(filters={'serviceTypes': ["serviceTypes"]}), True)
+))
+def test_should_identify_filter_search_from_query_params(query, expected):
+    return assert_equal, is_filtered(query), expected, query
```

Comment:
Bumping the dependency on pytest in this PR now causes the tests to throw hundreds of deprecation warnings about yield tests, so fixing that should be part of this story (Sorry). Whether or not we remove them as part of this PR or another one, I don't really mind. It probably makes sense to do it as part of this PR just so that it's done atomically. I don't think it's a particularly arduous process to switch them over. You can mostly do it with a few regex find and replaces.

<img width="1461" alt="screen shot 2017-07-19 at 07 36 42" src="https://user-images.githubusercontent.com/2920760/28353757-0ee8bb02-6c55-11e7-826c-ca02c67af374.png">